homeassistant/components/mobile_app/webhook.py

```diff
@@ -681,8 +686,10 @@ async def webhook_get_config(hass, config_entry, data):
 @validate_schema({vol.Required("tag_id"): cv.string})
 async def webhook_scan_tag(hass, config_entry, data):
     """Handle a fire event webhook."""
+    # Importing tag via hass.components in case it is overridden
+    # in a custom_components (custom_components.tag)
+    tag: TagProtocol = hass.components.tag  # pylint: disable=used-before-assignment
```

Comment:
@cdce8p I think this is a false positive from `pylint`

Reply:
I think it's because it is as the top level but I'm not sure.

Reply:
> @cdce8p I think this is a false positive from `pylint`

That's an interesting case. Tbh I had always assumed that **all** annotations are evaluated unless stringified. Since the file is missing the future import, the warning is emitted.

However, as I just learned myself annotations on local variables will just be ignored.
https://peps.python.org/pep-0526/#runtime-effects-of-type-annotations
https://stackoverflow.com/questions/41291892/variable-type-annotation-nameerror-inconsistency

So yeah, this is a false-positive, but you could also just add `from __future__ import annotations` to silence it.

Reply:
Opened: https://github.com/PyCQA/pylint/issues/7773